home/utils.py

```diff
@@ -631,3 +432,284 @@ def export_csv_content(queryset: PageQuerySet, response: HttpResponse) -> None:
+    whatsapp_title: str = ""
+    whatsapp_body: str = ""
+    messenger_title: str = ""
+    messenger_body: str = ""
+    viber_title: str = ""
+    viber_body: str = ""
+    translation_tag: str = ""
+    tags: str = ""
+    quick_replies: str = ""
+    triggers: str = ""
+    locale: str = ""
+    next_prompt: str = ""
+    image_link: str = ""
+    doc_link: str = ""
+    media_link: str = ""
+    related_pages: str = ""
+    tuple_of_extra_rows: tuple = ()
+
+    def __init__(
+        self,
+        page: ContentPage,
+        structure_string: str,
+        side_panel_message_number: int = 0,
+    ) -> None:
+        messaging_platform_messages = {
+            "whatsapp": self.__get_messages(page.whatsapp_body),
+            "viber": self.__get_messages(page.viber_body),
+            "messenger": self.__get_messages(page.messenger_body),
+        }
+
+        self.__set_messages(
+            page, side_panel_message_number, messaging_platform_messages
+        )
+        self.__set_structure(structure_string)
+
+        if side_panel_message_number == 0:
+            self.parent = self.__get_parent_page(page)
+            self.web_title = page.title
+            self.web_subtitle = page.subtitle
+            self.web_body = str(page.body)
+            self.whatsapp_title = page.whatsapp_title
+            self.messenger_title = page.messenger_title
+            self.viber_title = page.viber_title
+            self.translation_tag = str(page.translation_key)
+            self.tags = self.__format_list_from_query_set(page.tags.all())
+            self.quick_replies = self.__format_list_from_query_set(
+                page.quick_replies.all()
+            )
+            self.triggers = self.__format_list_from_query_set(page.triggers.all())
+            self.locale = str(page.locale)
+            self.related_pages = self.__format_list_from_query_set(
+                self.__get_related_pages(page)
+            )
+
+    def get_row(self):
+        """Returns object as an exportable row"""
+        return [
+            self.side_panel_column_1,
+            self.side_panel_column_2,
+            self.side_panel_column_3,
+            self.side_panel_column_4,
+            self.side_panel_column_5,
+            self.side_panel_message_number,
+            self.parent,
+            self.web_title,
+            self.web_subtitle,
+            self.web_body,
+            self.whatsapp_title,
+            self.whatsapp_body,
+            self.messenger_title,
+            self.messenger_body,
+            self.viber_title,
+            self.viber_body,
+            self.translation_tag,
+            self.tags,
+            self.quick_replies,
+            self.triggers,
+            self.locale,
+            self.next_prompt,
+            self.image_link,
+            self.doc_link,
+            self.media_link,
+            self.related_pages,
+        ]
+
+    def __format_list_from_query_set(self, unformatted_query: PageQuerySet) -> str:
```

Comment:
Is there a reason we're using these super private double underscore methods?

Reply:
Its just habit from previous jobs where the convention was to make anything that doesnt need to be accessed from outside the class a private method, would it be better to leave them public?

Reply:
Yeah private with a single underscore is fine, the double underscore is python's super private thing, which we don't usually use much. The docs go into it a bit here: https://docs.python.org/3/tutorial/classes.html#private-variables